src/modules/auth/components/RoleDropdown.js

```diff
@@ -0,0 +1,173 @@
+	// // sets 0.5s delay before triggering the search call when filtering through tennants
+	// useEffect(() => {
+	// 	if (timeoutRef.current !== null) {
+	// 		clearTimeout(timeoutRef.current);
+	// 	}
+	// 	timeoutRef.current = setTimeout(() => {
+	// 		timeoutRef.current = null;
+	// 		retrieveRoleList(tenantID);
+	// 	}, 500);
+	// }, [filter]);
+
+	// const handleFilter = (e) => {
+	// 	setLimit(5);
+	// 	setFilter(e.target.value);
+	// }
+
+	// fetch roles for Tenant dropdowns
+	const retrieveRoleList = async (tenantId) => {
+		let response;
+		let objCopy;
+		let id = tenantId;
+		let parameters = {f: filter, i: limit, exclude_global: true};
+		if (tenantId === undefined) {
+			id = '*';
+			parameters['exclude_global'] = false;
+			objCopy = {...tenantObj};
+		} else {
+			objCopy = selectedTenants.find(obj => obj._id === tenantId);
+		}
+		try {
+			response = await SeaCatAuthAPI.get(`/role/${id}`, {params: parameters});
+			let selectedTenantsCopy = [...selectedTenants];
+			let ten = {...objCopy, roles: response.data};
+			selectedTenantsCopy[idx] = ten;
+			setSelectedTenants(selectedTenantsCopy);
+			setCount(response.data.count);
+			setDisplayTenantRoles(response.data);
+		} catch (e) {
+			console.error(e);
+			props.app.addAlert("warning", `${t("BulkAssignmentContainer|Failed to fetch roles")}. ${e?.response?.data?.message}`, 30);
+		}
+	}
+
+	// adds role to the list of selected roles
+	const addRole = (roleId, index) => {
+		let tenants = [...selectedTenants];
+		let tenant = {...tenantObj};
+		if (roleId.substring(0, 1) === '*') {
+			Object.assign(tenant, {global: true })
+		};
+		//create a copy of displayed roles
+		let displayTenantRolesCopy = [...displayTenantRoles?.data];
+		// remove selected item from the copy of displayed roles
+		displayTenantRolesCopy.splice(index, 1);
+		//update state with modified copy
+		setDisplayTenantRoles({...displayTenantRoles, data: displayTenantRolesCopy});
+		if (tenant.selectedRole) {
+			Object.assign(tenant, {'selectedRole': [...tenant.selectedRole, roleId] });
+		} else {
+			Object.assign(tenant, {'selectedRole': [roleId] });
+		};
+		tenants[idx] = tenant;
+		setSelectedTenants(tenants);
+	};
+
+	// Determines tooltip messages
+	const message = useMemo(() => {
+		let msg = "";
+		if((tenantObj._id === "Global roles")) {
+			if (tenantObj?.selectedRole && (tenantObj.selectedRole.length > 0)) {
+				msg = t("BulkAssignmentContainer|Selected global roles will be assignes to/unassigned from the selected credentials");
+			} else {
+				msg = t("BulkAssignmentContainer|If you wish to assign/unassign global roles, select individual roles. Otherwise leave it as is");
+			}
+		} else if (tenantObj?.selectedRole && (tenantObj.selectedRole.length > 0)) {
+			msg = t("BulkAssignmentContainer|Tenant") + " '" + tenantObj._id + "' " + t("BulkAssignmentContainer|and selected roles will be assigned to/unassigned from selected credentials");
```

Comment:
@petrKavulok Can you wrap that the dynamic string to `{t("BulkAssignmentContainer|Tenant")} '${ tenantObj._id}' ${t("BulkAssignmentContainer|and selected roles will be assigned to/unassigned from selected credentials")}`